huaweicloud/services/cce/data_source_huaweicloud_cce_nodes.go

```diff
@@ -0,0 +1,260 @@
+							Computed: true,
+						},
+						"subnet_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ecs_group_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"key_pair": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"root_volume": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"size": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"volumetype": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"extend_params": {
+										Type:     schema.TypeMap,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								}},
+						},
+						"data_volumes": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"size": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"volumetype": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"extend_params": {
+										Type:     schema.TypeMap,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								}},
+						},
+						"billing_mode": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"server_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"private_ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceCceNodesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	cceClient, err := config.CceV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud CCE client : %s", err)
+	}
+
+	listOpts := nodes.ListOpts{
+		Uid:   d.Get("node_id").(string),
+		Name:  d.Get("name").(string),
+		Phase: d.Get("status").(string),
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		listOpts.Name = v.(string)
+	}
+
+	if v, ok := d.GetOk("node_id"); ok {
+		listOpts.Uid = v.(string)
+	}
+
+	if v, ok := d.GetOk("status"); ok {
+		listOpts.Phase = v.(string)
+	}
+
+	refinedNodes, err := nodes.List(cceClient, d.Get("cluster_id").(string), listOpts)
+
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve Nodes: %s", err)
+	}
+
+	if len(refinedNodes) < 1 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
```

Comment:
For list data-source, if the list response is empty, we can set a empty list to `nodes` attribute.
I think this return msg is unnecessary.

Reply:
I think a empty list is meanless. We should let the user know if the query has no result, in case the data source is referred in other resource.